mesonbuild/interpreter/type_checking.py

```diff
@@ -29,6 +29,7 @@
     from ..mesonlib import EnvInitValueType
 
     _FullEnvInitValueType = T.Union[EnvironmentVariables, T.List[str], T.List[T.List[str]], EnvInitValueType, str, None]
+    PkgConfigDefineType = T.Optional[T.Tuple[str, str]]
```

Comment:
> Pretty weird place to define this!

Where would be the best place?

> (Also weird to see it change from a list to a tuple. Not clear to me that this is a real problem.)

List is a DSL detail, it must be empty or exactly 2 items. In python world that translate better to an optional tuple. The actual reason I'm changing that type is because list is not hashable but tuples are, which allows using lru_cache.